relay-event-normalization/src/normalize/span/tag_extraction.rs

```diff
@@ -354,11 +355,60 @@ static SQL_TABLE_EXTRACTOR_REGEX: Lazy<Regex> = Lazy::new(|| {
     Regex::new(r#"(?i)(from|into|update)(\s|")+(?P<table>(\w+(\.\w+)*))(\s|")+"#).unwrap()
 });
 
-/// Returns the table in the SQL query, if any.
-///
-/// If multiple tables exist, only the first one is returned.
-fn sql_table_from_query(query: &str) -> Option<&str> {
-    extract_captured_substring(query, &SQL_TABLE_EXTRACTOR_REGEX)
+/// Used to replace placeholders (parameters) by dummy values such that the query can be parsed.
+static SQL_PLACEHOLDER_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"(?:\?+|\$\d+|%(?:\(\w+\))?s|:\w+)").unwrap());
+
+/// Returns a sorted, comma-separated list of SQL tables, if any.
+fn sql_tables_from_query(db_system: Option<&str>, query: &str) -> Option<String> {
```

Comment:
Is there any reason to add the structs, imports, etc inside the function?

Reply:
I wanted the Visitor to be a private type of the function, since it is the only one that uses it. But I agree that it's not great for readability, so I'll move them out of the function.

Reply:
If visibility is the only concern, what about encapsulating it in a module?